web-app/src/games/api.games.controller.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  Body,
+  Controller,
+  Post,
+  Put,
+  Get,
+  Param,
+  Req,
+  UseGuards,
+  NotFoundException,
+} from '@nestjs/common';
+import { Request } from 'express';
+import { Coordinates } from '../common/utils';
+import { GamesService, GameAndStatus } from './games.service';
+import { Game } from './game.entity';
+import { AuthGuard } from '@nestjs/passport';
+
+@UseGuards(AuthGuard('jwt'))
+@Controller('api/games')
+export class ApiGamesController {
+  constructor(private readonly gamesService: GamesService) { }
+
+  @Post('')
+  async create(
+    @Body() params: { size?: number },
+    @Req() req: Request
+  ): Promise<Game> {
+    return await this.gamesService.createNewGame(params.size, req.user["username"]);
+  }
+
+  @Get(':id')
+  async findOne(
+    @Param('id') id: number,
+    @Req() req: Request
+  ): Promise<GameAndStatus> {
+    const game = await this.gamesService.findGameById(id);
+    if (game === undefined) throw new NotFoundException();
+    return this.gamesService.getGameAndStatus(game, req.user["username"]);
+  }

```

Comment:
It would be nice to filter out some of the players data, such as the password hash which is still present in the response 😅 

(here is the output with POST, but we have the same prb with GET)

![2022-02-24 17_01_44-Clipboard](https://user-images.githubusercontent.com/14542336/155561360-4ba3fbd0-f236-4c14-9c49-e879e07fd773.png)

Reply:
Yes, I think there is something missing between the entity layer and the service layer to transform the data (and hide sensitive data for example).